crawl-ref/source/abyss.cc

```diff
@@ -1766,25 +1766,40 @@ struct corrupt_env
     }
 };
 
-static void _place_corruption_seed(const coord_def &pos, int duration)
+static void _place_corruption_seed(const coord_def &pos, int duration, bool is_monster_version = false)
 {
-    env.markers.add(new map_corruption_marker(pos, duration));
+    if (is_monster_version)
+    {
+        env.markers.add(new map_corruption_marker_mons(pos, duration));
+
+    } else //is player version
+    {
+        env.markers.add(new map_corruption_marker(pos, duration));
+    }
+
     // Corruption markers don't need activation, though we might
     // occasionally miss other unactivated markers by clearing.
     env.markers.clear_need_activate();
 }
 
-static void _initialise_level_corrupt_seeds(int power)
+static void _initialise_level_corrupt_seeds(int power, monster* mons = nullptr, bool is_monster_version = false)
 {
     const int low = power * 40 / 100, high = power * 140 / 100;
     const int nseeds = random_range(-1, min(2 + power / 110, 4), 2);
 
     const int aux_seed_radius = 4;
 
-    dprf("Placing %d corruption seeds (power: %d)", nseeds, power);
+    // The corruption centred on the player is free.
```

Comment:
This comment is in the wrong place, isn't it?

Reply:
Correct place but just unhelpful/legacy. I've updated it to be more relevant and useful, though this may be changed further as I split the function out following the next comment.